tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -479,6 +484,16 @@ def init(self):
             return RunnerManagerState.error()
 
         self.browser.update_settings(self.state.test)
+        if self.browser.pac is not None:
```

Comment:
This part should go into the specific Browser classes rather than directly in the executor; this specific implementation seems suitable to go in `WebDriverBrowser`, but other browsers will require a different approach. Also we can probably make it avoid using the HTTP server to load the pac, even if we use the substitution machinery, and can definitely avoid reading unless the file changed.